### PR TITLE
Cover more test/material tests with leak tracking.

### PR DESCRIPTION
**Pull request by ksokolovskyi**
### Description
This PR changes `testWidgets` by `testWidgetsWithLeakTracking` in number of `flutter/test/material` tests according to the https://github.com/dart-lang/leak_tracker/issues/134.

### Tests
- Updates `material/chip_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `material/chip_theme_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `material/dropdown_form_field_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `material/dropdown_menu_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `material/dropdown_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `material/elevated_button_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `material/persistent_bottom_sheet_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `material/snack_bar_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `material/tab_bar_theme_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `material/text_field_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `material/text_form_field_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `material/tooltip_test.dart` to use `testWidgetsWithLeakTracking`.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

### CODE REVIEW

**Comment by ksokolovskyi** [comment]
cc @polina-c

File: packages/flutter/test/material/text_field_test.dart [hunk 1]
```diff
@@ -2620,11 +2626,17 @@ void main() {
 
     expect(controller.selection.isCollapsed, true);
     expect(controller.selection.baseOffset, testValue.indexOf('i'));
+
+    // End gesture and skip the magnifier hide animation, so it can release
+    // resources.
+    await gesture.up();
+    await tester.pump();
+    await tester.pump(const Duration(milliseconds: 150));
```
Comment (polina-c) [inline packages/flutter/test/material/text_field_test.dart:2634]: The same

File: packages/flutter/test/material/text_field_test.dart [hunk 2]
```diff
@@ -2697,6 +2709,12 @@ void main() {
...
```
Comment (polina-c) [inline packages/flutter/test/material/text_field_test.dart:2717]: And here.

File: packages/flutter/test/material/text_field_test.dart [hunk 3]
```diff
@@ -12438,6 +12456,12 @@ void main() {
       textOffsetToPosition(tester, 0).dy,
       moreOrLessEquals(firstCharY - lineHeight, epsilon: 1),
     );
+
+    // End gesture and skip the magnifier hide animation, so it can release
+    // resources.
+    await gesture.up();
+    await tester.pump();
+    await tester.pump(const Duration(milliseconds: 150));
```
Comment (polina-c) [inline packages/flutter/test/material/text_field_test.dart:12464]: Here too.

**Review by polina-c — CHANGES_REQUESTED** [review]
Some comments.

**Review by polina-c — APPROVED** [review]

**Comment by ksokolovskyi** [comment]
@polina-c, thanks a lot for the review!

**Pull request merged**